src/util.js

```diff
@@ -247,7 +247,13 @@ export function doesNotTargetIFrame(name) {
 }
 
 export function findLinkFromClickTarget(target) {
-  return findClosestRecursively(target, "a[href]:not([target^=_]):not([download])")
+  const link = findClosestRecursively(target, "a[href], a[xlink\\:href]")
+
+  if (!link) return null
+  if (link.hasAttribute("download")) return null
+  if (link.hasAttribute("target") && link.target !== "_self") return null
+
+  return link
```

Comment:
Previously, Turbo used a pure CSS selector to find links: `"a[href]:not([target^=_]):not([download])"`

This worked for most standard HTML links, but it caused issues with:
  - Links inside SVG elements
  - Custom elements or shadow DOM links
  - Certain [target] values like "_self"

To fix this, we moved to a small JS-based selection function. It:
  - Walks up the DOM tree recursively
  - Handles shadow DOM hosts and slots
  - Matches both native <a> elements and custom link elements like <s-link> or <s-button>
  - Filters out links targeting iframes or downloads, while respecting "_self"

This approach is more robust for our use case and closely mimics real user clicks, while avoiding brittle CSS selectors.

Reply:
There's a subtle difference between this and the original. In the original, we'd only exclude elements that had an underscore-prefixed `target`, like `_blank` or `_top`. Now we'll exclude elements with any target value except `_self`.

As far as I know, this shouldn't matter. A non-keyword `target` should open the link in a new browsing context or specific frame, neither of which should use turbo for navigation. So I think we're good, I just wanted to point it out in case anyone else knows better!